yamale/schema/datapath.py

```diff
@@ -0,0 +1,48 @@
+from operator import getitem
+from functools import reduce
+from ..util import isstr
+# ABCs for containers were moved to their own module
+try:
+    from collections.abc import Mapping, Iterable
+except ImportError:
+    from collections import Mapping, Iterable
+
+def paths(data):
+    items = _getitems(data)
+    if not items:
+        return [DataPath()]
+
+    return [DataPath(key) + subpath
+            for key, subdata in items
+            for subpath in paths(subdata)]
+
+def _getitems(obj):
+    if isinstance(obj, Mapping):
+        return obj.items()
+    if isinstance(obj, Iterable) and not isstr(obj):
+        return enumerate(obj)
+
+class DataPath(object):
+
+    def __init__(self, key=[]):
```

Comment:
Can you modify this to not use a mutable default argument? https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments 

I realize there is no issue in the current code, but I'd like to ensure changes by others don't turn this into a future problem. Thanks!